src/services/utilities.ts

```diff
@@ -1868,15 +1868,18 @@ namespace ts {
         if (/^\d+$/.test(text)) {
             return text;
         }
+        // Editors can pass in undefined or empty string - we want to infer the preference in those cases.
+        const quotePreference = preferences.quotePreference || "auto";
         const quoted = JSON.stringify(text);
-        switch (preferences.quotePreference) {
-            case undefined:
+        switch (quotePreference) {
+            // TODO use getQuotePreference to infer the actual quote style.

```

Comment:
Is this TODO tracked by a bug?

Reply:
No, it may be a good idea though

Reply:
Probably tag the one above with the same ID.